p2p/conn/secret_connection.go

```diff
@@ -97,17 +96,11 @@ func MakeSecretConnection(conn io.ReadWriteCloser, locPrivKey crypto.PrivKey) (*
 	// Sort by lexical order.
 	loEphPub, hiEphPub := sort32(locEphPub, remEphPub)
 
-	hkdfInit := hkdf.New(hash, []byte("INIT_HANDSHAKE"), nil, []byte("TENDERMINT_SECRET_CONNECTION_TRANSCRIPT_HASH"))
-	if _, err := io.ReadFull(hkdfInit, hdkfState[:]); err != nil {
-		return nil, err
-	}
+	transcript := merlin.NewTranscript("TENDERMINT_SECRET_CONNECTION_TRANSCRIPT_HASH")
 
-	hkdfLoEphPub := hkdf.New(hash, loEphPub[:], hdkfState[:], []byte("TENDERMINT_SECRET_CONNECTION_TRANSCRIPT_HASH"))
-	if _, err := io.ReadFull(hkdfLoEphPub, hdkfState[:]); err != nil {
-		return nil, err
-	}
+	transcript.AppendMessage([]byte("EPHEMERAL_LOWER_PUBLIC_KEYS"), loEphPub[:])
```

Comment:
```suggestion
	transcript.AppendMessage([]byte("EPHEMERAL_LOWER_PUBLIC_KEY"), loEphPub[:])
```

?

Reply:
Should we make this byte slice a variable as well?

Reply:
can be a package level const

Reply:
probably okay to leave them inside []byte until they are used somewhere else

Reply:
or are they: https://github.com/uber-go/guide/blob/master/style.md#avoid-string-to-byte-conversion.

Reply:
good point @marbar3778 !